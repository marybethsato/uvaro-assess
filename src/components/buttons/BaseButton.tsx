import React from "react";

interface ButtonProps {
  children: React.ReactNode;
  onClick?: () => void;
  disabled?: boolean;
  className?: string;
}

const BaseButton = ({ children, className = "", ...props }: ButtonProps) => {
  return (
    <button
      className={`px-7 py-3 rounded-md cursor-pointer ${className}`}
      {...props}
    >
      {children}
    </button>
  );
};

export default BaseButton;
