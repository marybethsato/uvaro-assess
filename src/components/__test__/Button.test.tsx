// src/components/__tests__/BaseButton.test.tsx
import { fireEvent, render, screen } from "@testing-library/react";
import BaseButton from "../buttons/BaseButton";

describe("BaseButton component", () => {
  test("renders children text", () => {
    render(<BaseButton>Click me</BaseButton>);
    expect(screen.getByRole("button")).toHaveTextContent("Click me");
  });

  test("applies className prop", () => {
    render(<BaseButton className="bg-red-500">Red</BaseButton>);
    expect(screen.getByRole("button")).toHaveClass("bg-red-500");
  });

  test("calls onClick handler when clicked", () => {
    const handleClick = jest.fn<void, []>();
    render(<BaseButton onClick={handleClick}>Click</BaseButton>);
    fireEvent.click(screen.getByRole("button"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("does not call onClick when disabled", () => {
    const handleClick = jest.fn<void, []>();
    render(
      <BaseButton onClick={handleClick} disabled>
        Disabled
      </BaseButton>
    );
    const btn = screen.getByRole("button");
    expect(btn).toBeDisabled();
    fireEvent.click(btn);
    expect(handleClick).not.toHaveBeenCalled();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<BaseButton>Snapshot</BaseButton>);
    expect(asFragment()).toMatchSnapshot();
  });
});
