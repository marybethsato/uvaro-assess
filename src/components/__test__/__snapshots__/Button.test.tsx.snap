// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="px-7 py-3 rounded-md cursor-pointer "
  >
    Snapshot
  </button>
</DocumentFragment>
`;
